### PR TITLE
[python][scikit-learn] Fixes a bug that prevented using multiple eval_metrics in LGBMClassifier

### PR DESCRIPTION
**Pull request by gramirezespinoza**
This PR fixes a bug in the LGBMClassifier class. The code as-is breaks when eval_metric is a list of strings. This is meant to work (as referenced in the documentation) and does work on all derived classes of LGBMModel except LGBMClassifier. The issue is in the following code in the master branch:

https://github.com/microsoft/LightGBM/blob/1e2013a306c0ca719fa076a733796a6372e5b77e/python-package/lightgbm/sklearn.py#L785-L798

with the fix the following line works as expected (see tests in PR):

`gbm = lgb.LGBMClassifier(**params).fit(eval_metric=['fair', 'error'], **params_fit)`

### CODE REVIEW

File: tests/python_package_test/test_sklearn.py [hunk 1]
```diff
@@ -922,3 +922,14 @@ def test_continue_training_with_model(self):
         self.assertEqual(len(init_gbm.evals_result_['valid_0']['multi_logloss']), 5)
         self.assertLess(gbm.evals_result_['valid_0']['multi_logloss'][-1],
                         init_gbm.evals_result_['valid_0']['multi_logloss'][-1])
+
+    def test_eval_metrics_lgbmclassifier(self):
```
Comment (StrikerRUS) [inline tests/python_package_test/test_sklearn.py:926]: Will it make sense to include the content of this test into already existing huge unit test for metrics as an enhancement for the case of `... invalid multiclass metric is replaced with binary alternative ...`
https://github.com/microsoft/LightGBM/blob/2792923d35e7f3d392474ceffe2185c661047b2e/tests/python_package_test/test_sklearn.py#L503
https://github.com/microsoft/LightGBM/blob/2792923d35e7f3d392474ceffe2185c661047b2e/tests/python_package_test/test_sklearn.py#L749-L753

Or just please move this new test closer to that one.

Comment (gramirezespinoza) [reply tests/python_package_test/test_sklearn.py]: @StrikerRUS yes definitely makes sense. I've moved the test inside of `test_metrics` but placed it next to the section 

> non-default metric with multiple metrics in eval_metric

as it is highly related to that test. Thanks!

File: tests/python_package_test/test_sklearn.py [hunk 2]
```diff
@@ -540,6 +540,17 @@ def test_metrics(self):
         self.assertIn('l2', gbm.evals_result_['training'])
         self.assertIn('mape', gbm.evals_result_['training'])
 
+        # non-default metric with multiple metrics in eval_metric for LGBMClassifier
+        X_classification, y_classification = load_breast_cancer(True)
+        params_classification = {'n_estimators': 2, 'verbose': -1, 'objective': 'binary', 'metric': 'binary_logloss'}
+        params_fit_classification = {'X': X_classification, 'y': y_classification,
+                      'eval_set': (X_classification, y_classification), 'verbose': False}
```
Comment (StrikerRUS) [inline tests/python_package_test/test_sklearn.py:547]: Please fix one linting error:
```
./tests/python_package_test/test_sklearn.py:547:23: E128 continuation line under-indented for visual indent
```

Comment (gramirezespinoza) [reply tests/python_package_test/test_sklearn.py]: Fixed!

Comment (StrikerRUS) [reply tests/python_package_test/test_sklearn.py]: Thank you! Sorry for the inconvenience, but could you please rebase to the latest `master`? I've just fixed R-package builds: #3225. Those failures prevents from merging.

Comment (gramirezespinoza) [reply tests/python_package_test/test_sklearn.py]: Rebased, hopefully I did it correctly as it is my first time rebasing an upstream branch 😄

Comment (StrikerRUS) [reply tests/python_package_test/test_sklearn.py]: Thanks!
But seems that something went wrong: see number of changed files

image

Maybe you can allow editing from maintainers and then I'll be able to help with cleaning up the rebasing?
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Comment (gramirezespinoza) [reply tests/python_package_test/test_sklearn.py]: Screenshot 2020-07-14 at 22 34 14
 editing for maintainers is enabled. Sorry for this.

Comment (StrikerRUS) [reply tests/python_package_test/test_sklearn.py]: No problem! Seems I was able to exclude excess files.
